src/UKHO.ADDS.EFS.Builder.S100/Pipelines/Assemble/DownloadFilesNode.cs

```diff
@@ -1,10 +1,117 @@
+using UKHO.ADDS.EFS.Builder.S100.Pipelines.Assemble.Logging;
+using UKHO.ADDS.Infrastructure.Pipelines;
 using UKHO.ADDS.Infrastructure.Pipelines.Nodes;
+using UKHO.ADDS.Infrastructure.Results;
 
 namespace UKHO.ADDS.EFS.Builder.S100.Pipelines.Assemble
 {
     internal class DownloadFilesNode : ExchangeSetPipelineNode
     {
-        protected override Task<NodeResultStatus> PerformExecuteAsync(IExecutionContext<ExchangeSetPipelineContext> context) => Task.FromResult(NodeResultStatus.NotRun);
+        private readonly IFileShareReadOnlyClient _fileShareReadOnlyClient;
+        private ILogger _logger;
+
+        private const long FileSizeInBytes = 10485760;
+        private const string ProductName = "ProductName";
+        private const string EditionNumber = "EditionNumber";
+        private const string UpdateNumber = "UpdateNumber";
+
+        public DownloadFilesNode(IFileShareReadOnlyClient fileShareReadOnlyClient)
+        {
+            _fileShareReadOnlyClient = fileShareReadOnlyClient ?? throw new ArgumentNullException(nameof(fileShareReadOnlyClient));
+        }
+
+        protected override async Task<NodeResultStatus> PerformExecuteAsync(IExecutionContext<ExchangeSetPipelineContext> context)
+        {
+            try
+            {
+                var batchDetails = context.Subject.BatchDetails;
+                var downloadPath = Path.Combine(context.Subject.WorkSpaceRootPath, "fssdata");
+
+                _logger = context.Subject.LoggerFactory.CreateLogger<DownloadFilesNode>();
+
+                EnsureDownloadDirectoryExists(downloadPath);
+
+                var result = await DownloadLatestBatchFilesAsync(context, SelectLatestBatchesByProductEditionAndUpdate(batchDetails));
+
+                return result;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogDownloadFilesNodeFailed(ex.Message);
+
+                return NodeResultStatus.Failed;
+            }
+        }
+
+        private static IEnumerable<BatchDetails> SelectLatestBatchesByProductEditionAndUpdate(IEnumerable<BatchDetails> batchDetails)
+        {
+            var latestBatches = batchDetails
+                .Where(b => b.Attributes != null)
+                .GroupBy(b =>
+                    new
+                    {
+                        ProductName = b.Attributes.FirstOrDefault(a => a.Key == ProductName)?.Value,
+                        EditionNumber = b.Attributes.FirstOrDefault(a => a.Key == EditionNumber)?.Value,
+                        UpdateNumber = b.Attributes.FirstOrDefault(a => a.Key == UpdateNumber)?.Value
+                    })
+                .Select(g => g.OrderByDescending(b => b.BatchPublishedDate).First())
+                .ToList();
+
+            return latestBatches;
+        }
+
+        private async Task<NodeResultStatus> DownloadLatestBatchFilesAsync(IExecutionContext<ExchangeSetPipelineContext> context, IEnumerable<BatchDetails> latestBatches)
```

Comment:
Instead of passing ExchangeSetPipelineContext can we pass the required parameters. 
Same applies for LogFssDownloadFailed.
```suggestion
private async Task<NodeResultStatus> DownloadLatestBatchFilesAsync(IEnumerable<BatchDetails> latestBatches, string workSpaceRootPath, string correlationId)
```